src/branching.jl

```diff
@@ -0,0 +1,74 @@
+"""
+    branch!(tree, node)
+
+Get the branching variable with [`get_branching_variable`](@ref) and then calls [`get_branching_nodes_info`](@ref) and [`add_node!`](@ref).
+"""
+function branch!(tree, node)
+    variable_idx = get_branching_variable(tree, tree.options.branch_strategy, node)
+    # no branching variable selected => return
+    variable_idx == -1 && return 
+    nodes_info = get_branching_nodes_info(tree, node, variable_idx)
+    add_node!.(tree, nodes_info)
```

Comment:
```suggestion
    for node_info in nodes_info
        add_node!(tree, nodes_info)
    end
```
otherwise the broadcast allocates a vector of nothing

Reply:
Shall I rather have a `add_nodes!` function then? I thought about that before and then thought: Ah why not use broadcast :smile: